Adding `deferInitialization` docs. Fixes #1304.

filmaj opened the pull request:
I will definitely need a review on the Japanese docs since I just ran the text through Google Translate 😅

srajiang commented on docs/_advanced/deferring_initialization.md:
```diff
@@ -0,0 +1,30 @@
+(async () => {
+  await app.init();
+  // Now safe to call start()
+  await app.start();
```
```suggestion
  await app.start(process.env.PORT || 3000);
```

srajiang commented on docs/_advanced/deferring_initialization.md:
```diff
@@ -0,0 +1,30 @@
+---
+title: Deferring initialization
```
```suggestion
title: Deferring App initialization
```

seratch replied:
Good suggestion!

srajiang reviewed (approved):
👍 Englais looks good to me. Very minor comments!

seratch commented on docs/_advanced/deferring_initialization.md:
```diff
@@ -0,0 +1,30 @@
+
+// Must call init() before start() within an async function
+(async () => {
+  await app.init();
```
Can you add try/catch clause to synchronously catch he error and terminate the process? https://github.com/slackapi/bolt-js/pull/1303/files#r799876413 the exist status should not be necessarily 255, though. `process.exist(1)` would be fine too.

seratch commented on docs/_advanced/ja_deferring_initialization.md:
```diff
@@ -0,0 +1,30 @@
+---
+title: 初期化の延期
```
It's already understandable! 🤣 but @wongjas and I can suggest updates once we finalize the English version

wongjas commented on docs/_advanced/deferring_initialization.md:
```diff
@@ -0,0 +1,30 @@
+---
+
+<div class="section-content">
+Bolt offers a way to defer full initialization via the `deferInitialization` option and to call the equivalent `App#init()` in your code, putting more control over asynchronous execution required for initialization into your hands as the developer.
```
Is the syntax here supposed to be `App#init()` and not `App.init()`? I'm not too familiar with JavaScript.

seratch replied:
@wongjas `#` here means an instance method in documents - it's not an actual syntax in code. If you say `App.init()`, that sounds like a static method of the class.

filmaj commented:
Thanks for the suggestions! I've added them and updated the PR. If there are no other suggestions for the English docs, then perhaps the Japanese docs could get a pass to try to go from "understandable" to "good" 😄 ?

seratch commented:
@filmaj Thanks for revising the doc quickly! @wongjas will come up with JP translation parts today (I've already reviewed most of his translation)

wongjas commented on docs/_advanced/ja_deferring_initialization.md:
```diff
@@ -0,0 +1,35 @@
+---
+
+<div class="section-content">
+Boltは、`deferInitialization` オプションを介して完全な初期化を延期し、コード内で同等の `App＃init（）` を呼び出す方法を提供し、開発者として初期化に必要な非同期実行をより詳細に制御します。
```
```suggestion
Bolt は `deferInitialization` というオプションを使うことで、アプリの初期化処理の完了を遅延させることができます。このオプションを使う場合、あなたのコードの中で遅延された初期化処理部分に対応する `App#init()` メソッドを呼び出す必要がありますが、こうすることで初期化に必要となる非同期処理の実行をよりコントールすることが可能となります。
```

seratch replied:
Sorry, this is my mistake - s/コントール/コントロール/

wongjas commented on docs/_advanced/ja_deferring_initialization.md:
```diff
@@ -0,0 +1,35 @@
+---
+title: 初期化の延期
```
```suggestion
title: アプリの初期化の遅延
```

wongjas commented on docs/_advanced/ja_deferring_initialization.md:
```diff
@@ -0,0 +1,35 @@
+<div class="section-content">
+Boltは、`deferInitialization` オプションを介して完全な初期化を延期し、コード内で同等の `App＃init（）` を呼び出す方法を提供し、開発者として初期化に必要な非同期実行をより詳細に制御します。
+
+_注：`init（）` の前に `start（）` を呼び出すと、Boltは例外を発生させます。_
```
```suggestion
_注意: `init()` メソッドを呼び出す前に `start()` メソッドを呼び出した場合、 Bolt は例外を発生させます。_
```

wongjas commented on docs/_advanced/ja_deferring_initialization.md:
````diff
@@ -0,0 +1,35 @@
+```javascript
+const { App } = require('@slack/bolt');
+
+// オプションとして、コンストラクタで deferInitialization を設定可能
````
```suggestion
// deferInitialization はコンストラクターで指定できるオプション
```

wongjas commented on docs/_advanced/ja_deferring_initialization.md:
```diff
@@ -0,0 +1,35 @@
+
+(async () => {
+  try {
+    // 非同期関数内で`start（）`の前に`init（）`を呼び出す必要があります
```
```suggestion
    // 非同期関数の中で start() メソッドを呼び出す前に init() メソッドを呼び出すこと
```

wongjas commented on docs/_advanced/ja_deferring_initialization.md:
```diff
@@ -0,0 +1,35 @@
+  try {
+    // 非同期関数内で`start（）`の前に`init（）`を呼び出す必要があります
+    await app.init();
+    // これで、`start（）`を安全に呼び出すことができます
```
```suggestion
    // init() メソッドを呼び出したので、`start()` メソッドを安全に呼び出すことができる
```

wongjas commented on docs/_tutorials/ja_reference.md:
```diff
@@ -116,6 +116,7 @@ App オプションは、`App` のコンストラクターに渡します。
 | `clientOptions.slackApiUrl` | Slack Web API で使用するエンドポイントをカスタマイズできます。これが使用されるのはほとんどがテスト用途です。 | 
 | `socketMode` | 真偽値を指定するオプションで、`true` に設定するとアプリは[ソケットモード](/bolt-js/ja-jp/concepts#socket-mode)で起動します。ソケットモードは WebSocket のコネクションを通して Slack からのデータを受信する機能です。デフォルトは `false` です。
 | `developerMode` | デベロッパーモードを有効にする真偽値です。 `true` に設定したとき、`logLevel` は `DEBUG`、 `socketMode` は `true` に自動的に設定されます。しかし、 これらの二つのプロパティを明示的に設定した場合、それらの設定が `developerMode` による設定よりも優先されます。さらに、デバッグをしやすくするためのカスタムの OAuth エラーハンドラーも提供されます。また、全ての Slack からのリクエストのボディがログ出力されるため、トークンのようなセンシティブな情報がログに含まれる可能性があります。デフォルトは `false` です。| 
+| `deferInitialization` | アプリの初期化を延期し、開発者に非同期`App＃init（）`メソッドを手動で呼び出す責任を負わせるブール値。 `init（）`は、`App＃start（）`の前に呼び出す必要があります。 デフォルトは`false`です。 |
```
```suggestion
| `deferInitialization` | アプリの初期化を遅延させる真偽値です。有効にすると非同期の `App#init()` メソッドを手動で呼び出す必要があります。 また `init()` メソッドは `App#start()` を実行する前に呼び出さなければなりません。 デフォルトは `false` です。 |
```

wongjas commented:
Added the Japanese translation suggestions. One final check please @seratch! 🙇

seratch commented:
@wongjas I might forget submitting your review - I cannot see your suggestions

seratch reviewed (commented):
@wongjas one correction (it was my fault, sorry)

filmaj commented:
Thanks for the suggestions! I've added and squashed them. @seratch or @wongjas, feel free to merge if you think the Japanese documentation is now ready 👍

seratch reviewed (approved):
LGTM but the timing to merge this PR wold be after the v3.10 package release.

srajiang commented:
v3.10 is released! 🎉 
Rebased and merging this now.
